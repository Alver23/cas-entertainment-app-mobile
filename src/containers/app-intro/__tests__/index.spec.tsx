// Dependencies
import React, { Dispatch } from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import AppIntroSlider from 'react-native-app-intro-slider';

// Components
import { AppIntro } from '../index';

// Mocks
import itemMocks from '../components/item/__tests__/mocks.json';

describe('AppIntro container', () => {
  let component: ShallowWrapper;

  const mockRef = { current: { goToSlide: jest.fn() } };

  const props = {
    onSkip: (): jest.Mock => jest.fn(),
  };

  const mockState: [unknown, Dispatch<Function>] = [1, jest.fn()];

  beforeEach(() => {
    jest.spyOn(React, 'useState').mockReturnValueOnce(mockState);
    jest.spyOn(React, 'useRef').mockReturnValue(mockRef);
    component = shallow(<AppIntro {...props} />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call the onSlidechange and nextSlide callback', () => {
    const { onSlideChange, renderNextButton, renderDoneButton, renderItem } = component.find(AppIntroSlider).props();
    onSlideChange();
    renderNextButton().props.onPress();
    expect(mockState[1]).toHaveBeenCalled();
    expect(mockRef.current.goToSlide).toHaveBeenCalled();
    expect(renderDoneButton().props).toEqual(
      expect.objectContaining({
        title: expect.any(String),
      }),
    );
    expect(renderItem({ item: { ...itemMocks.item } }).props).toEqual(
      expect.objectContaining({
        item: expect.any(Object),
      }),
    );
  });
});
