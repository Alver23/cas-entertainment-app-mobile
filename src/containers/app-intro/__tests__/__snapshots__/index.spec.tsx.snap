// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIntro container should rendering correctly 1`] = `
<AppIntroSlider
  activeDotStyle={
    Object {
      "backgroundColor": undefined,
    }
  }
  bottomButton={true}
  data={
    Array [
      Object {
        "image": Object {
          "testUri": "../../../src/containers/app-intro/assets/one.png",
        },
        "key": "one",
        "text": "Lorem ipsum dolor sit amet.",
        "title": "PRACTISE",
      },
      Object {
        "image": Object {
          "testUri": "../../../src/containers/app-intro/assets/two.png",
        },
        "key": "two",
        "text": "Lorem ipsum dolor sit amet.",
        "title": "HEALTHY",
      },
      Object {
        "image": Object {
          "testUri": "../../../src/containers/app-intro/assets/three.png",
        },
        "key": "three",
        "text": "Lorem ipsum dolor sit amet.",
        "title": "EVERYDAY",
      },
    ]
  }
  doneLabel="Done"
  dotClickEnabled={true}
  dotStyle={
    Object {
      "backgroundColor": undefined,
    }
  }
  nextLabel="Next"
  onSkip={[Function]}
  onSlideChange={[Function]}
  prevLabel="Back"
  renderDoneButton={[Function]}
  renderItem={[Function]}
  renderNextButton={[Function]}
  showDoneButton={true}
  showNextButton={true}
  showPrevButton={false}
  showSkipButton={true}
  skipLabel="Skip"
/>
`;
